cheatsheets/Cryptographic_Storage_Cheat_Sheet.md

```diff
@@ -1,234 +1,185 @@
+- At the application level.
+- At the database level (e.g, [SQL Server TDE](https://docs.microsoft.com/en-us/sql/relational-databases/security/encryption/transparent-data-encryption?view=sql-server-ver15)
+- At the filesystem level (e.g, BitLocker or LUKS)
+- At the hardware level (e.g, encrypted RAID cards or SSDs)
 
-Ensure that the implementation has (at minimum) had some cryptography experts involved in its creation. If possible, use an implementation that is FIPS 140-2 certified.
+Which layer(s) are most appropriate will depend on the threat model. For example, hardware level encryption is effective at protecting against the physical theft of the server, but will provide no protection if an attacker is able to compromise the server remotely.
 
-See [NIST approved algorithms](http://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt1r4.pdf) Table 2 "Comparable strengths" for the strength ("security bits") of different algorithms and key lengths, and how they compare to each other.
+### Minimise the Storage of Sensitive Information
 
-- In general, where different algorithms are used, they should have comparable strengths e.g. if an AES-128 key is to be encrypted, an AES-128 key or greater, or RSA-3072 or greater could be used to encrypt it.
-- In general, hash lengths are twice as long as the security bits offered by the symmetric/asymmetric algorithm  e.g. SHA-224 for 3TDEA (112 security bits) (due to the [Birthday Attack](http://en.wikipedia.org/wiki/Birthday_attack))
+The best way to protect sensitive information is to not store it in the first place. Although this applies to all kinds of information, it is most often applicable to credit card details, as they are highly desirable for attackers, and PCI DSS has such stringent requirements for how they must be stored. Wherever possible, the storage of sensitive information should be avoided.
 
-If a password is being used to protect keys then the [password strength](http://en.wikipedia.org/wiki/Password_strength) should be sufficient for the strength of the keys it is protecting.
+## Algorithms
 
-When 3DES is used, ensure `K1 != K2 != K3`, and the minimum key length must be `192 bits` .
+For symmetric encryption **AES** with a key that's at least **128 bits** (ideally **256 bits**) and a secure [mode](#cipher-modes) should be used as the preferred algorithm.
 
-#### Rule - Use approved cryptographic modes
+For asymmetric encryption, use elliptical curve cryptography (ECC) with a secure curve such as **X25519** as a preferred algorithm. If ECC is not available, **RSA** must be used with a key that is at least **2048 bits** in size.
```

Comment:
Change 'X25519' to 'Curve25519' to fit proper naming convention or change sentence structure. The Curve primitive is called 'Curve25519', the DH function 'X25519', and the EdDSA function 'Ed25519'.

Reply:
Fixed in e95c53c.